kubetest2-aks/deployer/up.go

```diff
@@ -56,6 +59,70 @@ type UpOptions struct {
 	K8sVersion       string `flag:"k8sVersion" desc:"--k8sVersion flag for cluster Kubernetes version"`
 }
 
+type enableCustomFeaturesPolicy struct{}
+
+type changeLocationPolicy struct{}
+
+type addCustomConfigPolicy struct {
+	customConfig string
+}
+
+func (p enableCustomFeaturesPolicy) Do(req *policy.Request) (*http.Response, error) {
+	req.Raw().Header.Add("AKSHTTPCustomFeatures", "Microsoft.ContainerService/EnableCloudControllerManager")
+	return req.Next()
+}
+
+func (p changeLocationPolicy) Do(req *policy.Request) (*http.Response, error) {
+	if strings.Contains(req.Raw().URL.String(), "Microsoft.ContainerService/locations") {
+		q := req.Raw().URL.Query()
+		q.Set("api-version", "2017-08-31")
+		req.Raw().URL.RawQuery = q.Encode()
+	}
+
+	return req.Next()
+}
+
+func (p addCustomConfigPolicy) Do(req *policy.Request) (*http.Response, error) {
```

Comment:
Since I have another PR https://github.com/kubernetes-sigs/cloud-provider-azure/pull/2877 to address the same issue, this change is more elegant, I am going to close https://github.com/kubernetes-sigs/cloud-provider-azure/pull/2877
Thanks for your fix.

Reply:
I think your comments in PR and descriptions are helpful. Do you think they can be merged?

Reply:
Yeah, please.